blog/_src/posts/2017-02-21-bullets-are-good-for-your-coq-proofs.md

```diff
@@ -0,0 +1,238 @@
+    Title: Bullets are good for your Coq proofs
+    Date: 2017-02-21T19:04:28
+    Tags: coq, by Gabriel Scherer
+
+I believe that bullets are one of the most impactful features of
+recent versions of Coq, among those that non-super-expert users can
+enjoy. They had a big impact on the maintainability of my
+proofs. Unfortunately, they are not very well-known, due to the fact
+that some introductory documents have not been updated to use them.
+
+Bullets are a very general construction and there are several possible
+ways to use them; I have iterated through different styles. In this
+post I will give the general rules, and explain my current usage
+style.
+
+<!-- more -->
+
+# Why bullets
+
+While you are doing a proof, Coq shows a list of subgoals that have to
+be proved before the whole proof is complete. Most proof steps will
+operate on the current active subgoal, changing the hypotheses or the
+goal to prove, but some proof steps will split it into several
+subgoals (growing the total list of goals), or may terminate the proof
+of the current subgoal and show you the next active subgoal.
+
+Before bullets, a typical proof script would contain the proofs of
+each subgoal, one after another.
+
+```
+induction foo. (* this creates many subgoal *)
+
+proof of first subgoal.
+
+proof of second subgoal.
+```
+
+There are many ways to structure this to make the structure more
+apparent: people would typically have a comment on each subgoal, or
+make disciplined use of indentation and blank lines. But, in my
+experience, a major problem with this style was maintenability in the
+face of changes to the definitions or parts of automation. It could be
+very hard of what was happening when a proof suddenly broke after
```

Comment:
❗️  I don't understand this line. Is the purpose of this sentence just to say "For example:" ?